ank/src/cli_commands/set_state.rs

```diff
@@ -117,9 +156,207 @@ impl CliCommands {
 //////////////////////////////////////////////////////////////////////////////
 #[cfg(test)]
 mod tests {
-    use std::io;
+    use super::{
+        create_state_with_default_workload_specs, io, overwrite_using_field_mask, process_inputs,
+        CliCommands, StoredWorkloadSpec,
+    };
+    use crate::cli_commands::server_connection::MockServerConnection;
+    use api::ank_base::UpdateStateSuccess;
+    use common::{
+        objects::{CompleteState, RestartPolicy, State},
+        state_manipulation::Object,
+    };
+    use mockall::predicate::eq;
+    use serde_yaml::Value;
+    use std::{collections::HashMap, io::Cursor};
 
     pub fn read_to_string_mock(_file: String) -> io::Result<String> {
-        Ok("".into())
+        Ok(_file)
+    }
+
+    const RESPONSE_TIMEOUT_MS: u64 = 3000;
+
+    const SAMPLE_CONFIG: &str = r#"desiredState:
+        workloads:
+          nginx:
+            agent: agent_A
+            tags:
+            - key: owner
+              value: Ankaios team
+            dependencies: {}
+            restartPolicy: ALWAYS
+            runtime: podman
+            runtimeConfig: |
+              image: docker.io/nginx:latest
+              commandOptions: ["-p", "8081:80"]"#;
+
+    // [utest->swdd~cli-provides-set-desired-state~1]
+    #[test]
+    fn utest_create_state_with_default_workload_specs_empty_update_mask() {
+        let update_mask = vec![];
+
+        let complete_state = create_state_with_default_workload_specs(&update_mask);
+
+        assert!(complete_state.desired_state.workloads.is_empty());
+    }
+
+    // [utest->swdd~cli-provides-set-desired-state~1]
+    #[test]
+    fn utest_create_state_with_default_workload_specs_with_update_mask() {
+        let update_mask = vec![
+            "desiredState.workloads.nginx.restartPolicy".to_string(),
+            "desiredState.workloads.nginx2.restartPolicy".to_string(),
+            "desiredState.workloads.nginx3".to_string(),
+        ];
+
+        let complete_state = create_state_with_default_workload_specs(&update_mask);
+
+        assert!(complete_state.desired_state.workloads.contains_key("nginx"));
+        assert!(complete_state
+            .desired_state
+            .workloads
+            .contains_key("nginx2"));
```

Comment:
We can make this test more strict to also consider the expected values:
```suggestion
        assert_eq!(
            complete_state.desired_state.workloads.get("nginx"),
            Some(&StoredWorkloadSpec::default())
        );

        assert_eq!(
            complete_state.desired_state.workloads.get("nginx2"),
            Some(&StoredWorkloadSpec::default())
        );
```